### DISCUSSION TITLE
Issue with Uploading Signing Certificates in Codemagic - "Cannot save Signing Certificates without certificate private key" Error

### DISCUSSION THREAD

**Discussion by lukyanenk7** [codemagic-ci-cd/codemagic-docs#2356] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

iOS (Swift, Objective-C)

### Steps to reproduce

![Знімок екрана 2024-02-04 о 15 36 37](https://github.com/codemagic-ci-cd/codemagic-docs/assets/142221278/9c56a913-51d0-4a3f-b559-bf2a86768f4f)

I am encountering an issue during the CI/CD process on the Codemagic platform and need assistance. Throughout my build process, I am continuously receiving an error stating, "Cannot save Signing Certificates without certificate private key". I have ensured that all necessary development and distribution certificates (Development_Certificate and Distribution_Certificate), as well as the provisioning profiles (provisioning_profiles), have been uploaded to Codemagic and appear to be configured correctly.

Despite numerous attempts and checks, I have been unable to identify the source of the problem. The certificates and provisioning profiles are visible in my Codemagic account and match those I use in Xcode and App Store Connect. The private key for the certificates has also been added to Codemagic.

Could I have missed an important step in setting up my CI/CD process? Does anyone have suggestions on how I can resolve this issue, or can anyone point out what I might be doing wrong?

Any help would be greatly appreciated.

### Expected results

I expect that after successfully uploading the development and distribution certificates (Development_Certificate and Distribution_Certificate), as well as the provisioning profiles (provisioning_profiles), to Codemagic, the CI/CD process should complete without errors. Specifically, the "Fetch signing files" step should successfully retrieve the necessary files and the build should proceed to the next steps without the error "Cannot save Signing Certificates without certificate private key".

### Actual results

Despite having uploaded the necessary certificates and provisioning profiles to Codemagic, and confirming that they are correctly set up in the codemagic.yaml file, the build process fails. The error "Cannot save Signing Certificates without certificate private key" is consistently returned during the "Fetch signing files" step. This indicates that Codemagic is unable to find or use the private key associated with the uploaded certificates, even though I have ensured that the private key is present and correctly configured in Codemagic.

### Build id (optional)

65bf9260e81f1d3e012b8fa7

**Comment by dtrdic** [comment]
Hello @lukyanenk7 please make sure you are not base64 encoding your private key. Add it without encoding it

**Reply by dtrdic** [reply]
hey @lukyanenk7 i wanted to follow up on your issue. 
Just to give some additional information, as i see that you indeed have uploaded the certificate and the provisioning profile to Codemagic platform, but the thing that is missing is a configuration inside yaml that references that provisioning profile and which will fetch all uploaded signing files matching a specific distribution type and bundle identifier during the build, you will need to define the `distribution_type` and `bundle_identifier` fields in your codemagic.yaml configuration. So when defining the bundle identifier `com.Sweet-puzzle-2`, Codemagic will fetch any uploaded certificates and profiles matching this extension com.Sweet-puzzle-2
```
    ios-workflow:
      name: iOS Workflow
      instance_type: mac_mini_m1
      max_build_duration: 120
      integrations:
        app_store_connect: Codemagic API key 
      environment:
        ios_signing:
          distribution_type: developement
          bundle_identifier: com.Sweet-puzzle-2
```
 also note that if you want to publish to the App Store you will need to create a distribution provisioning profile and then set the distribution_type to `app_store` in your yaml
```
         ios_signing:
          distribution_type: app_store
          bundle_identifier: com.Sweet-puzzle-2
```

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
